web/api/v1/api.go

```diff
@@ -1516,7 +1547,23 @@ func (api *API) rules(r *http.Request) apiFuncResult {
 		if len(apiRuleGroup.Rules) > 0 {
 			rgs = append(rgs, apiRuleGroup)
 		}
+
+		if paginationRequest != nil && len(rgs) == int(paginationRequest.MaxRuleGroups+1) {
```

Comment:
Hm, if we move this check to the beginning of the loop, we potentially can drop the (perhaps surprising) +1 for this and removal of that one element in the end and simplifying the end part of the API body
```
	if paginationRequest != nil {
		if len(rgs) == int(paginationRequest.MaxRuleGroups+1) {
			nextRg := rgs[paginationRequest.MaxRuleGroups]
			res.NextToken = getRuleGroupNextToken(nextRg.File, nextRg.Name)
			res.RuleGroups = rgs[:paginationRequest.MaxRuleGroups]
		} else {
			res.RuleGroups = rgs
		}
		return apiFuncResult{res, nil, nil, nil}
	}
	res.RuleGroups = rgs
	return apiFuncResult{res, nil, nil, nil}
```

Reply:
great suggestion. I'm not sure If I coded exactly what you asked, but I got the idea. I put a separated commit just for this so that it is simpler to review.